src/lib.rs

```diff
@@ -1,4 +1,779 @@
+        self.read_struct(region.addr(), ptr::addr_of_mut!(dos_hdr))?;
+        if dos_hdr.e_magic != IMAGE_DOS_SIGNATURE {
+            return None;
+        }
+
+        // Read the FileHeader header
+        let nt_header_addr = region.addr() + dos_hdr.e_lfanew as usize;
+        let file_header_addr = nt_header_addr + offset_of!(IMAGE_NT_HEADERS, FileHeader);
+        let mut file_header: IMAGE_FILE_HEADER = unsafe { mem::zeroed() };
+        self.read_struct(file_header_addr, ptr::addr_of_mut!(file_header))?;
+
+        // Get the appropriate NT header structure
+        let nt_header_data = self.read(
+            nt_header_addr,
+            match file_header.Machine {
+                IMAGE_FILE_MACHINE_AMD64 => mem::size_of::<IMAGE_NT_HEADERS64>(),
+                IMAGE_FILE_MACHINE_I386 => mem::size_of::<IMAGE_NT_HEADERS32>(),
+                _ => return None,
+            },
+        )?;
+
+        let nt_header: Box<dyn PeImage> = match file_header.Machine {
+            IMAGE_FILE_MACHINE_AMD64 => unsafe {
+                Box::new(mem::transmute_copy::<
+                    [u8; mem::size_of::<IMAGE_NT_HEADERS64>()],
+                    IMAGE_NT_HEADERS64,
+                >(&nt_header_data[..].try_into().ok()?))
+            },
+            IMAGE_FILE_MACHINE_I386 => unsafe {
+                Box::new(mem::transmute_copy::<
+                    [u8; mem::size_of::<IMAGE_NT_HEADERS32>()],
+                    IMAGE_NT_HEADERS32,
+                >(&nt_header_data[..].try_into().ok()?))
+            },
+            _ => return None,
+        };
+
+        if !nt_header.valid_signature() {
+            return None;
+        }
+
+        Some(nt_header)
+    }
+
+    pub fn iter_images(
+        &self,
+    ) -> impl Iterator<Item = (MEMORY_BASIC_INFORMATION, Box<dyn PeImage + '_>)> {
+        self.iter_regions().filter_map(|m| {
+            if !m.is_image() {
+                return None;
+            }
+
+            match self.read_image_header(&m) {
+                Some(img) => Some((m, img)),
+                _ => None,
+            }
+        })
+    }
+}
+
+impl Drop for Process {
+    fn drop(&mut self) {
+        unsafe {
+            if !self.h.is_null() {
+                CloseHandle(self.h);
+            }
+        }
+    }
+}
+
+fn find_ntdll(proc: &Process, machine: u16) -> Option<usize> {
+    proc.iter_images().find_map(|(m, img)| {
+        if !img.is_dll() || img.machine() != machine {
+            return None;
+        }
+
+        let exp_dd = img.export_dd()?;
+        let mut exp_dir: IMAGE_EXPORT_DIRECTORY = unsafe { mem::zeroed() };
+        proc.read_struct(
+            m.addr() + exp_dd.VirtualAddress as usize,
+            ptr::addr_of_mut!(exp_dir),
+        )?;
+
+        let exp_name_addr = m.addr() + exp_dir.Name as usize;
+        match proc.read(exp_name_addr, "ntdll.dll\0".chars().count()) {
+            Some(name) if name == b"ntdll.dll\0" => Some(m.addr()),
```

Comment:
```suggestion
        match proc.read(exp_name_addr, "ntdll.dll\0".len()) {
            Some(name) if name.eq_ignore_ascii_case(b"ntdll.dll\0") => Some(m.addr()),
```
Slightly safer, if in by some weird chance the dll filename isn't fully lowercase?

Reply:
Done